Library/Homebrew/extend/pathname.rb

```diff
@@ -529,6 +529,12 @@ def unlink
     ObserverPathnameExtension.n += 1
   end
 
+  def mkpath
+    return if exist?
+    super
+    puts "mkdir #{self}" if ARGV.verbose?
+  end
```

Comment:
I see what you're saying. That looks not quite equivalent, it'd have to be something like:
```
pathExisted = exist?
super
puts "mkdir #{self}" if ARGV.verbose? && !pathExisted
```
because the goal is to log directory creation if the directory didn't exist.
I can change it to that if you prefer, but it seems more complicated.

In fact, to make it even simpler, I'd maybe log the action unconditionally, and log `mkdir -p` so that the log keeps being a valid Unix script.
```
def mkpath
    super
    puts "mkdir -p #{self}" if ARGV.verbose?
  end
```

And on reflection, this should be `ObserverPathnameExtension.verbose?` like everything here, not `ARGV.verbose?`, to follow the other refactorings.